econml/dml.py

```diff
@@ -696,6 +696,7 @@ def fit(self, Y, T, X=None, W=None, sample_weight=None, sample_var=None, inferen
         if sample_var is not None and inference is not None:
             warn("This estimator does not yet support sample variances and inference does not take "
                  "sample variances into account. This feature will be supported in a future release.")
+        Y, T, X, W, sample_weight, sample_var = check_input_arrays(Y, T, X, W, sample_weight, sample_var)
```

Comment:
Why only add the check for SparseLinearDML, but not the other classes in this file?

Reply:
Because the sparse methods call other methods (e.g. `check_high_dimensional`) before calling the super's fit. All other DML and DR estimators call the parent's fit (`_ortho_learner` in this case).